libs/model/src/main/java/de/cyface/model/MetaData.kt

```diff
@@ -0,0 +1,180 @@
+    /**
+     * The type of device uploading the data, such as "Pixel 3" or "iPhone 6 Plus".
+     *
+     * The setter is required by Apache Flink.
+     */
+    @JvmField
+    @set:Suppress("unused")
+    var deviceType: String? = null
+    /**
+     * The operating system version, such as "Android 9.0.0" or "iOS 11.2".
+     *
+     * The setter is required by Apache Flink.
+     */
+    @JvmField
+    @set:Suppress("unused")
+    var osVersion: String? = null
+    /**
+     * The version of the app that transmitted the measurement, such as "1.2.0" or "1.2.0-beta1".
+     *
+     * The setter is required by Apache Flink.
+     */
+    @JvmField
+    @set:Suppress("unused")
+    var appVersion: String? = null
+    /**
+     * The length of the measurement in meters.
+     *
+     * The setter is required by Apache Flink.
+     */
+    @JvmField
+    @set:Suppress("unused")
+    var length: Double = 0.0
+    /**
+     * The id of the user who has uploaded this measurement.
+     *
+     * The setter is required by Apache Flink.
+     */
+    @JvmField
+    @set:Suppress("unused")
+    var userId: UUID? = null
+    /**
+     * The format version in which the `Measurement` was deserialized, e.g. "1.2.3".
+     *
+     * The setter is required by Apache Flink.
+     */
+    @JvmField
+    var version: String? = null
+    /**
+     * The upload date when the `Measurement` was uploaded to the collector.
+     *
+     * The setter is required by Apache Flink.
+     */
+    @JvmField
+    var uploadDate: Date? = null
+
+    /**
+     * Creates new completely initialized `MetaData`.
+     *
+     * @param identifier The worldwide unique identifier of the measurement.
+     * @param deviceType The type of device uploading the data, such as "Pixel 3" or "iPhone 6 Plus".
+     * @param osVersion The operating system version, such as "Android 9.0.0" or "iOS 11.2".
+     * @param appVersion The version of the app that transmitted the measurement, such as "1.2.0" or "1.2.0-beta1".
+     * @param length The length of the measurement in meters.
+     * @param userId The id of the user who has uploaded this measurement.
+     * @param version The format version in which the `Measurement` was deserialized, e.g. "2.0.0".
+     * @param uploadDate The upload date when the `Measurement` was uploaded to the collector.
+     */
+    constructor(
+        identifier: MeasurementIdentifier?,
+        deviceType: String,
+        osVersion: String,
+        appVersion: String,
+        length: Double,
+        userId: UUID,
+        version: String,
+        uploadDate: Date,
+    ) {
+        require(version.matches(SUPPORTED_VERSIONS.toRegex())) { "Unsupported version: $version" }
+        this.identifier = identifier
+        this.deviceType = deviceType
+        this.osVersion = osVersion
+        this.appVersion = appVersion
+        this.length = length
+        this.userId = userId
+        this.version = version
+        this.uploadDate = uploadDate
+    }
```

Comment:
I guess this is auto-converted. Would be easier to just declare all the properties directly in the header 
```kotlin
data class MetaData(var identifier: MeasurementIdentifier, ...): Serializable {
init {
  require(version.matches(SUPPORTED_VERSIONS.toRegex())) { "Unsupported version: $version" }
}
```

Reply:
see above, it now looks similar, but had to drop the data class and add a create factory which calls validate, due to the no-arg requirement